src/protocol/inav_nrf24l01.c

```diff
@@ -0,0 +1,814 @@
+    NRF24L01_WriteReg(NRF24L01_04_SETUP_RETR, NRF24L01_04_ARD_2500us | NRF24L01_04_ARC_1); // one retry after 2500us
+    NRF24L01_WriteReg(NRF24L01_05_RF_CH, RF_CHANNEL_BIND);
+    // bitrate and power are set using regigister NRF24L01_06_RF_SETUP
+    NRF24L01_SetBitrate(NRF24L01_BR_250K);
+    NRF24L01_SetPower(Model.tx_power);
+    // Writing to the STATUS register clears the specified interrupt bits
+    NRF24L01_WriteReg(NRF24L01_07_STATUS, BV(NRF24L01_07_RX_DR) | BV(NRF24L01_07_TX_DS) | BV(NRF24L01_07_MAX_RT));
+    // NRF24L01_08_OBSERVE_TX is read only register
+    // NRF24L01_09_RPD is read only register (called RPD for NRF24L01+, CD for NRF24L01)
+    NRF24L01_WriteRegisterMulti(NRF24L01_0A_RX_ADDR_P0, rx_tx_addr_bind, RX_TX_ADDR_LEN); // RX_ADDR_P0 must equal TX_ADDR for auto ACK
+    // RX_ADDR for pipes P1-P5 (registers 0B to 0F) aret left at default values
+    NRF24L01_WriteRegisterMulti(NRF24L01_10_TX_ADDR, rx_tx_addr_bind, RX_TX_ADDR_LEN);
+    NRF24L01_WriteReg(NRF24L01_11_RX_PW_P0, payload_size);
+    // Payload Widths for P1-P5 (registers 12 to 16) aret left at default values
+    // NRF24L01_17_FIFO_STATUS is read only register for all non-reserved bits
+    // No registers with values 18 to 1B
+#ifdef USE_AUTO_ACKKNOWLEDGEMENT
+    NRF24L01_ReadReg(NRF24L01_1D_FEATURE);
+    NRF24L01_Activate(0x73); // Activate feature register, needed for NRF24L01 (harmless for NRF24L01P)
+    NRF24L01_ReadReg(NRF24L01_1D_FEATURE);
+    NRF24L01_WriteReg(NRF24L01_1D_FEATURE, BV(NRF24L01_1D_EN_ACK_PAY) | BV(NRF24L01_1D_EN_DPL));
+    NRF24L01_WriteReg(NRF24L01_1C_DYNPD, BV(NRF24L01_1C_DYNPD_P0)); // dynamic payload length on pipes P0
+#endif
+
+    // Check for Beken BK2421/BK2423 chip
+    // It is done by using Beken specific activate code, 0x53
+    // and checking that status register changed appropriately
+    // There is no harm to run it on nRF24L01 because following
+    // closing activate command changes state back even if it
+    // does something on nRF24L01
+    NRF24L01_Activate(0x53); // magic for BK2421 bank switch
+    dbgprintf("Trying to switch banks\n");
+    if (NRF24L01_ReadReg(NRF24L01_07_STATUS) & 0x80) {
+        dbgprintf("BK2421 detected\n");
+        // Beken registers don't have such nice names, so we just mention
+        // them by their numbers
+        // It's all magic, eavesdropped from real transfer and not even from the
+        // data sheet - it has slightly different values
+        NRF24L01_WriteRegisterMulti(0x00, (u8 *) "\x40\x4B\x01\xE2", 4);
+        NRF24L01_WriteRegisterMulti(0x01, (u8 *) "\xC0\x4B\x00\x00", 4);
+        NRF24L01_WriteRegisterMulti(0x02, (u8 *) "\xD0\xFC\x8C\x02", 4);
+        NRF24L01_WriteRegisterMulti(0x03, (u8 *) "\x99\x00\x39\x21", 4);
+        NRF24L01_WriteRegisterMulti(0x04, (u8 *) "\xF9\x96\x82\x1B", 4);
+        NRF24L01_WriteRegisterMulti(0x05, (u8 *) "\x24\x06\x7F\xA6", 4);
+        NRF24L01_WriteRegisterMulti(0x06, (u8 *) "\x00\x00\x00\x00", 4);
```

Comment:
Please remove initialization of Beken registers 0x06 -0x0b with zeros. It's an artifact of decoding of sloppy initialization code from some early TX.